protocols/data_processor.py

```
@@ -4,101 +4,105 @@
+            "action": "data_processing",
+            "mode": "simulation",
+            "message": "No data directory found, returning simulated results",
+            "files_processed": 3,
+            "total_records": 150,
+            "insights": [
+                "Simulated: Found 3 data files",
+                "Simulated: Processed 150 records total",
+                "Simulated: Average processing time 0.5s per file",
             ],
-            'timestamp': datetime.utcnow().isoformat()
+            "timestamp": datetime.utcnow().isoformat(),
```

Ruff (0.11.9)

`datetime.datetime.utcnow()` used

(DTZ003)